airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/BigQueryAvroSerializedBuffer.java

```diff
@@ -52,7 +51,7 @@ public static BufferCreateFunction createBufferFunction(final S3AvroFormatConfig
     return (pair, catalog) -> {
       final AirbyteStream stream = catalog.getStreams()
           .stream()
-          .filter(s -> s.getStream().getName().equals(pair.getName()) && StringUtils.equals(s.getStream().getNamespace(), pair.getNamespace()))
```

Comment:
Is it true that even in the async world, we shouldn't have more than one namespace? Like even for CDC syncs, its multiple streams/table which are all being written to the same namespace? Does this logic have to consider any of the name mangling logic in the catalog parser?

Reply:
git says you might be the most qualified to answer that 😅 https://github.com/airbytehq/airbyte/commit/db28b13ed43ebab7666273dfaa5a26d8e1528393#diff-d68a5be057956e8fb021561f4c7111aa1e72eac6b87906eee8d7058a99f87163

but we explicitly support connections with two streams that have the same name but different namespace (iirc there's a DAT that verifies this...)

> name mangling

depends on how exactly we're calling into this class, but StreamId has `originalNamespace` / `originalName` for this exact reason - as long as we're only using the mangled names inside the sql generator, it should be fine. (I haven't verified that we're doing it correctly though)